### PR TITLE
Add tutorial for Heartbeat plugin

### PR DESCRIPTION
**Pull request by sbis04**
I have added the tutorial for the `Heartbeat` plugin. The page contents are as follows:

Heartbeat plugin

Tag #891

### CODE REVIEW

**Review by proudofsimin — APPROVED** [review]
You are so great as always!

File: docusaurus/docs/using-plugin-with-wechaty/heartbeat.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+:::
+
+## Adding Heartbeat plugin
+
+For the demonstration of adding this plugin, we will use the **Starter Bot** and show you how to add the `Heartbeat` plugin to it. Follow the steps below:
+
+### 1. Create a starter bot
+
+Follow the instructions on the **Starter Bot** page to create the foundation of a Wechaty bot.
```
Comment (lijiarui) [inline docusaurus/docs/using-plugin-with-wechaty/heartbeat.md]: I thinks we should add link for **Starter Bot* page and then users can refer that part if he don't know how to build a starter bot.

File: docusaurus/docs/using-plugin-with-wechaty/heartbeat.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+Import the plugin inside the `starter-bot.ts` file:
+
+```ts
+import { Heartbeat } from 'wechaty-plugin-contrib'
+```
+
+Define a variable called `config` where you can specify:
+
+* `contact`: The contact of the person whom you want to send the emoji (default: **filehelper**)
+* `emoji`: Define under `heartbeat`, the emoji to send (default: **爱心**)
```
Comment (lijiarui) [inline docusaurus/docs/using-plugin-with-wechaty/heartbeat.md]: change  `default: **爱心**` to `[爱心] - Heartbeat emoji`

File: docusaurus/docs/using-plugin-with-wechaty/heartbeat.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+const config = {
+    contact: 'cotact_num@c.us', // contact who will receive the emoji
+    emoji: {
+        heartbeat: '😎', // the emoji to send
+    },
+    intervalSeconds: 60, // sends the emoji after an interval of 60 seconds
+}
+```
+
+For using with **WhatsApp** you can set the contact in the following format:
```
Comment (lijiarui) [inline docusaurus/docs/using-plugin-with-wechaty/heartbeat.md]: If you try to introduce WhatsApp here, then I suggest you introduce Wechat and WeCom here.

File: docusaurus/docs/using-plugin-with-wechaty/heartbeat.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+Define a variable called `config` where you can specify:
+
+* `contact`: The contact of the person whom you want to send the emoji (default: **filehelper**)
+* `emoji`: Define under `heartbeat`, the emoji to send (default: **爱心**)
+* `intervalSeconds`: The time interval before sending an emoji (default: **1 hour**)
+
+```ts
+const config = {
+    contact: 'cotact_num@c.us', // contact who will receive the emoji
```
Comment (lijiarui) [inline docusaurus/docs/using-plugin-with-wechaty/heartbeat.md]: I suggest this using default contact `filehelper` 

`filehelper` is suitable for Wechat puppet, `cotact_num@c.us` is suitable for Whatsapp puppet.

File: docusaurus/docs/using-plugin-with-wechaty/heartbeat.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+)
+```
+
+> You will get more information regarding the `QRCodeTerminal` plugin in the tutorial [here](./qr-code-terminal).
+
+### 5. Run the bot
+
+To run the bot, first you have to **export/set** an environment variable with the type of puppet to use, and then start the bot:
+
+#### Linux/macOS
```
Comment (lijiarui) [inline docusaurus/docs/using-plugin-with-wechaty/heartbeat.md]: I think we should unify the style here, @shwetalsoni did a good job, it looks like this：

image

You can use code like this:

```markdown
<Tabs
  groupId="operating-systems"
  defaultValue="linux"
  values={[
    { label: 'Linux',   value: 'linux', },
    { label: 'macOS',   value: 'mac', },
    { label: 'Windows', value: 'windows', },
  ]
}>

<TabItem value="linux">

```

File: docusaurus/docs/using-plugin-with-wechaty/heartbeat.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+![Heartbeat plugin output](../../static/img/docs/using-plugin-with-wechaty/heartbeat/heartbeat-bot-output.png)
+
+Congratulations! You have successfully integrated the `Heartbeat` plugin to your Wechaty bot.
+
+## Conclusion
+
+You can apply a similar concept to add the `Heartbeat` plugin to any of your Wechaty bots. You can learn more about this plugin [here](https://github.com/wechaty/wechaty-plugin-contrib#4-heartbeat).
+
+## References
+
```
Comment (lijiarui) [inline docusaurus/docs/using-plugin-with-wechaty/heartbeat.md]: Add video link reference here: https://www.youtube.com/watch?v=tfGZXoe_aA4

**Review by lijiarui — CHANGES_REQUESTED** [review]
Pls change as I request, thanks!

**Comment by lijiarui** [comment]
ping @sbis04

**Comment by lijiarui** [comment]
Pls resolve the conflict @sbis04

File: docusaurus/docs/using-plugin-with-wechaty/heartbeat.md (hunk 4 above)
Comment (lijiarui) [reply docusaurus/docs/using-plugin-with-wechaty/heartbeat.md]: @sbis04 It still show as `cotact_num@c.us` here, I think we should change to `filehelper`

**Comment by sbis04** [comment]
@lijiarui Have made the CI green.

File: docusaurus/docs/using-plugin-with-wechaty/heartbeat.md (hunk 6 above)
Comment (sbis04) [reply docusaurus/docs/using-plugin-with-wechaty/heartbeat.md]: Have added the video link

File: docusaurus/docs/using-plugin-with-wechaty/heartbeat.md (hunk 5 above)
Comment (sbis04) [reply docusaurus/docs/using-plugin-with-wechaty/heartbeat.md]: Have added the tabs

File: docusaurus/docs/using-plugin-with-wechaty/heartbeat.md (hunk 4 above)
Comment (sbis04) [reply docusaurus/docs/using-plugin-with-wechaty/heartbeat.md]: Have updated it now

File: docusaurus/docs/using-plugin-with-wechaty/heartbeat.md (hunk 3 above)
Comment (sbis04) [reply docusaurus/docs/using-plugin-with-wechaty/heartbeat.md]: As I have updated with `filehelper`, I have removed the reference to WhatsApp.

File: docusaurus/docs/using-plugin-with-wechaty/heartbeat.md (hunk 2 above)
Comment (sbis04) [reply docusaurus/docs/using-plugin-with-wechaty/heartbeat.md]: Have updated it.

File: docusaurus/docs/using-plugin-with-wechaty/heartbeat.md (hunk 1 above)
Comment (sbis04) [reply docusaurus/docs/using-plugin-with-wechaty/heartbeat.md]: Yes, I'll add the link. Currently, the **Starter Bot** page isn't there on the docs, we have it further down on our proposal timeline. Once, the page is ready we will also update the link here.

**Comment by sbis04** [comment]
@lijiarui I have made the suggested changes

File: docusaurus/docs/using-plugin-with-wechaty/heartbeat.md (hunk 1 above)
Comment (lijiarui) [reply docusaurus/docs/using-plugin-with-wechaty/heartbeat.md]: @sbis04 Could you file a new issue for this and we can trace this later, then I will ignore this part.

**Review by lijiarui — APPROVED** [review]
LGTM.

And also looking forward for you to file a new issue for the pending part.

**Comment by lijiarui** [comment]
@sbis04 Since the `.webp` format has lots of advantages and the [safari has supported it last year](https://www.macrumors.com/2020/06/22/webp-safari-14/) which means all the modern browsers will be able to compatible with it.

@huan has just added a new unit test for enforcing the image format: the CI will not be passed if we are using other image formats than the perfect .webp.

Please make CI green to adjust the new rule, see https://github.com/wechaty/wechaty.js.org/issues/1035

**Comment by sbis04** [comment]
@lijiarui I have converted the image to `.webp` format and made the CI green.

**Review by huan — APPROVED** [review]

**Pull request merged**